subjects/traversal.py

```diff
@@ -1,3 +1,4 @@
+from django.db import IntegrityError
```

Comment:
Need a couple blank lines after this import, which is why flake8 is complaining in the automated tests :)